### ISSUE TITLE
DataStore Query iOS / Android discrepancies for boolean type

### ISSUE THREAD

**Issue by Lorenzohidalgo** [aws-amplify/amplify-flutter#752]
**Describe the bug**
DataStore does not handle the equal operator equally for boolean variables in Android and iOS.

For Android the query needs to pass the boolean as a string:
```dart
final _result = await Amplify.DataStore.query(
        Members.classType,
        where: Members.SPACEPID
            .eq(spacePID)
            .and(Members.USERPID.eq(userPID))
            .and(Members.ISDELETED.eq('false')),
      );
```

For iOS the query will needs to pass the boolean as a boolean to work correctly:
```dart
final _result = await Amplify.DataStore.query(
        Members.classType,
        where: Members.SPACEPID
            .eq(spacePID)
            .and(Members.USERPID.eq(userPID))
            .and(Members.ISDELETED.eq(false)),
      );
```

Why does this happen?

**Expected behavior**
Both platforms should use the same operator logic.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[X] Android
[X] iOS

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5 20G71 darwin-arm, locale en-ES)
    • Flutter version 2.2.3 at /Users/lhidalgo/development/flutter
    • Framework revision f4abaa0735 (4 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/lhidalgo/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • C5B88706-5F4E-45FC-AE2F-9FD6941AA666 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 91.0.4472.164
```

</details>

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro Max (simulator)
 - OS: iOS 14.5

**Additional context**

**Comment by HuiSF** [comment]
Thanks for reporting this issue. I took a quick looking on query predicate serialization and deserialization between Flutter and amplify-android. It converts boolean value to 0 (if the provided boolean is `false`) and 1 (for `true`). This seems correct as SQLite doesn't have a separate Boolean class so boolean field will be stored as 0 or 1.

Then I inspected SQLite DB using Android Studio DB inspector, the boolean field value is stored as string... it seems the issue. Will investigate further and update.

**Comment by HuiSF** [comment]
The SQLite table indeed has wrong type `Text` set in Android for the `published` column that I set as boolean. The type is correct as `Integer` in iOS.

<img width="562" alt="Screen Shot 2021-07-26 at 4 41 53 PM" src="https://user-images.githubusercontent.com/10602282/127072901-c54aa9e6-3423-4c83-b6b4-ae1b4eb857ac.png">

**Comment by Lorenzohidalgo** [comment]
Thanks @HuiSF, this other issue might be also another platform discrepancy #511, since @offlineprogrammer is able to query by `.eq(null)` but it will fail on Android.

**Comment by Jordan-Nelson** [comment]
**Note:** Prior to this issue being marked as resolved, the DataStore integration tests that are skipped due to this issue should be updated. 

See: https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/example/integration_test/query_test/query_predicate_test/bool_query_predicate_test.dart

**Comment by HuiSF** [comment]
Hi @Lorenzohidalgo the fix for this bug is testable now with [0.3.0-rc.1](https://pub.dev/packages/amplify_datastore/versions/0.3.0-rc.1/changelog). This is a breaking change, please try it out, and follow up if you encounter any issues.
